schemas/bedrock-edv-storage.js

```
@@ -250,7 +250,162 @@ const query = {
+          type: 'array',
+          minItems: 1,
+          items: {type: 'string'}
+        }]
+      },
+      controller: {
+        title: 'controller',
+        type: 'string'
+      },
+      delegator: {
+        anyOf: [{
+          type: 'string'
+        }, {
+          type: 'array',
+          minItems: 1,
+          items: {type: 'string'}
+        }]
+      },
+      invoker: {
+        anyOf: [{
+          type: 'string'
+        }, {
+          type: 'array',
+          minItems: 1,
+          items: {type: 'string'}
+        }]
+      },
+      invocationTarget: {
+        title: 'Invocation Target',
+        anyOf: [{
+          type: 'string'
+        }, {
+          type: 'object',
+          properties: {
+            id: {
+              title: 'Invocation Target Id',
+              type: 'string'
+            },
+            type: {
+              title: 'Invocation Target Type',
+              type: 'string'
+            }
+          }
+        }]
+      },
+      parentCapability: {
+        title: 'Parent Capability',
+        type: 'string'
+      },
+      proof: {
+        type: 'object',
+        additionalProperties: false,
+        properties: {
+          verificationMethod: {
+            title: 'verificationMethod',
+            type: 'string'
+          },
+          type: {
+            title: 'type',
+            type: 'string'
+          },
+          created: {
+            title: 'created',
+            type: 'string'
+          },
+          proofPurpose: {
+            title: 'proofPurpose',
+            type: 'string'
+          },
+          capabilityChain: {
+            title: 'capabilityChain',
+            type: 'array',
+            minItems: 1,
+            items: {type: 'string'}
+          },
+          jws: {
+            title: 'jws',
+            type: 'string'
+          },
+        }},
+      referenceId: {
+        title: 'Reference Id',
+        type: 'string'
+      }
+    }},
+  {
+    anyOf: [
```

I recommend changing the order of this `allOf` so that this simple required check is first and therefore it will fail fast if the required fields are not present.

---

Makes sense, updated!